lib/ui/questions/survey_questions_view_model.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:async';
+
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey_flutter_ic/di/provider/di.dart';
+import 'package:survey_flutter_ic/model/survey_details_model.dart';
+import 'package:survey_flutter_ic/ui/details/survey_details_ui_model.dart';
+import 'package:survey_flutter_ic/ui/questions/survey_questions_view_state.dart';
+import 'package:survey_flutter_ic/usecase/base/base_use_case.dart';
+import 'package:survey_flutter_ic/usecase/get_survey_details_use_case.dart';
+
+final surveyQuestionsViewModelProvider = StateNotifierProvider.autoDispose<
+        SurveyQuestionsViewModel, SurveyQuestionsViewState>(
+    (_) => SurveyQuestionsViewModel(
+          getIt.get<GetSurveyDetailsUseCase>(),
+        ));
+
+final surveyDetailsStream = StreamProvider.autoDispose<SurveyDetailsUiModel>(
+    (ref) => ref
+        .watch(surveyQuestionsViewModelProvider.notifier)
+        ._surveyDetailsStreamController
+        .stream);
+
+final visibleIndexStream = StreamProvider.autoDispose<int>((ref) => ref
+    .watch(surveyQuestionsViewModelProvider.notifier)
+    ._visibleIndexStreamController
+    .stream);
+
+final nextQuestionStream = StreamProvider.autoDispose<void>((ref) => ref
+    .watch(surveyQuestionsViewModelProvider.notifier)
+    ._nextQuestionStreamController
+    .stream);
+
+class SurveyQuestionsViewModel extends StateNotifier<SurveyQuestionsViewState> {
+  final GetSurveyDetailsUseCase _getSurveyDetailsUseCase;
+
+  final _surveyDetailsStreamController =
+      StreamController<SurveyDetailsUiModel>();
+  final _visibleIndexStreamController = StreamController<int>();
+  final _nextQuestionStreamController = StreamController<void>();
+
+  SurveyQuestionsViewModel(this._getSurveyDetailsUseCase)
+      : super(const SurveyQuestionsViewState.init());
+
+  void setVisibleSurveyIndex(int index) {
+    _visibleIndexStreamController.add(index);
+  }
+
+  void nextQuestion() {
+    _nextQuestionStreamController.add(null);
+  }
+
+  Future getSurveyDetails(String surveyId) async {
+    state = const SurveyQuestionsViewState.loading();
+    final result = await _getSurveyDetailsUseCase.call(surveyId);
+
+    if (result is Failed<SurveyDetailsModel>) {
+      final error = result.getErrorMessage();
+      state = SurveyQuestionsViewState.error(error);
+    } else {
+      final surveyDetails = result as Success<SurveyDetailsModel>;
+      _surveyDetailsStreamController
+          .add(SurveyDetailsUiModel.fromModel(surveyDetails.value));
+      state = const SurveyQuestionsViewState.success();
```

Comment:
@hoangnguyen92dn I think the code can be optimized a bit.
We can take advantage of `SurveyQuestionsViewState` by obtaining `SurveyDetailsUiModel` to `success` state, like:
- In `survey_questions_view_state.dart`: modify to `const factory SurveyQuestionsViewState.success(SurveyDetailsUiModel surveyDetailsUiModel) = _Success;`
- Then we can set the state to:
`state = SurveyQuestionsViewState.success(SurveyDetailsUiModel.fromModel(surveyDetails.value))`
- In `survey_questions_screen.dart`:
```dart
Widget build(BuildContext context) {
    final state = ref.watch(surveyQuestionsViewModelProvider);
    return Scaffold(
      body: state.maybeWhen(
        loading: () => _buildLoadingIndicator(),
        success: (surveyDetailUiModel) => _buildSurveyQuestionsContent(surveyDetailUiModel),
        orElse: () => const SizedBox.shrink(),
      ),
    );
  }
```

Reply:
@manh-t It was also my thought 😄 But I created a Stream for the survey details to handle the data outside the ViewState. Let me update 🆗